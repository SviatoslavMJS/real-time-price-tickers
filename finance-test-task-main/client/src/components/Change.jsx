import ArrowDown from "./ArrowDown";
import ArrowUp from "./ArrowUp";
import s from '../styles/Change.module.css';

function Change(props) {
    return (
        <div className={props.vector ? s.green : s.red}>
            { props.vector ? <ArrowUp /> : <ArrowDown />}
            {props.change}
        </div>
    )
}

export default Change;