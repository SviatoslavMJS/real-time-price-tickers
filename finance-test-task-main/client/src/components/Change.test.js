import renderer from 'react-test-renderer';
import Change from './Change';


test('On change prop vector in Change class must change from green to red / child component from ArrowUp to ArrowDown', () => {
    const component = renderer.create(
      <Change change={"111"} vector={true}/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.vector = false;

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.vector = true;

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    
});

test('On change prop change in Change text must change from 111 to 999', () => {
    const component = renderer.create(
      <Change change={"111"} vector={true}/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.change = "999";

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.change = "111";

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    
});