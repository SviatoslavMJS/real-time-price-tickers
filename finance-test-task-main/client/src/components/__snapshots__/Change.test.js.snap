// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`On change prop change in Change text must change from 111 to 999 1`] = `
<div
  className="green"
>
  <svg
    className="svg_up"
  >
    <path
      d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
    />
  </svg>
  111
</div>
`;

exports[`On change prop change in Change text must change from 111 to 999 2`] = `
<div
  className="green"
>
  <svg
    className="svg_up"
  >
    <path
      d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
    />
  </svg>
  111
</div>
`;

exports[`On change prop change in Change text must change from 111 to 999 3`] = `
<div
  className="green"
>
  <svg
    className="svg_up"
  >
    <path
      d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
    />
  </svg>
  111
</div>
`;

exports[`On change prop vector in Change class must change from green to red / child component from ArrowUp to ArrowDown 1`] = `
<div
  className="green"
>
  <svg
    className="svg_up"
  >
    <path
      d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
    />
  </svg>
  111
</div>
`;

exports[`On change prop vector in Change class must change from green to red / child component from ArrowUp to ArrowDown 2`] = `
<div
  className="green"
>
  <svg
    className="svg_up"
  >
    <path
      d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
    />
  </svg>
  111
</div>
`;

exports[`On change prop vector in Change class must change from green to red / child component from ArrowUp to ArrowDown 3`] = `
<div
  className="green"
>
  <svg
    className="svg_up"
  >
    <path
      d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
    />
  </svg>
  111
</div>
`;
